components/director/internal/tenantfetcher/service.go

```diff
@@ -197,15 +201,49 @@ func (s Service) SyncTenants() error {
 	return nil
 }
 
-func (s Service) createTenants(ctx context.Context, currTenants map[string]bool, eventsTenants []model.BusinessTenantMappingInput) error {
+func (s Service) createTenants(ctx context.Context, currTenants map[string]string, eventsTenants []model.BusinessTenantMappingInput) error {
 	tenantsToCreate := make([]model.BusinessTenantMappingInput, 0)
+	parentsToCreate := make([]model.BusinessTenantMappingInput, 0)
 	for i := len(eventsTenants) - 1; i >= 0; i-- {
-		if currTenants[eventsTenants[i].ExternalTenant] {
+		if _, ok := currTenants[eventsTenants[i].ExternalTenant]; ok {
 			continue
 		}
+
+		if len(eventsTenants[i].Parent) > 0 {
+			if parentInternalID, ok := currTenants[eventsTenants[i].Parent]; ok {
+				eventsTenants[i].Parent = parentInternalID
+			} else {
+				parentTenant := model.BusinessTenantMappingInput{
+					Name:           eventsTenants[i].Parent,
+					ExternalTenant: eventsTenants[i].Parent,
+					Subdomain:      "",
+					Parent:         "",
+					Type:           tenant.TypeToStr(tenant.Customer),
+					Provider:       s.providerName,
+				}
+				parentsToCreate = append(parentsToCreate, parentTenant)
+			}
+		}
 		tenantsToCreate = append(tenantsToCreate, eventsTenants[i])
 	}
 
+	parentsToCreate = s.dedupeTenants(parentsToCreate)
```

Comment:
I'm wondering how we can reduce the lines of code here, and extracting lines 230-239 in a `createParents` method - it can also return a map if the creates tenants